src/sexpr.rs

```diff
@@ -279,6 +321,11 @@ impl<'a> std::fmt::Display for DisplayExpr<'a> {
         fn fmt_sexpr(f: &mut std::fmt::Formatter, arena: &Arena, sexpr: SExpr) -> std::fmt::Result {
             match arena.get(sexpr) {
                 SExprData::Atom(data) => std::fmt::Display::fmt(data, f),
+                SExprData::String(data) => {
+                    write!(f, "\"")?;
+                    std::fmt::Display::fmt(data, f)?;
+                    write!(f, "\"")
```

Comment:
This code could be slightly simplified, and better handle escaped characters with

```suggestion
                    std::fmt::Debug::fmt(data, f)?;
```